core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/workflow/Workflow.java

```diff
@@ -1,6 +1,26 @@
 package uk.ac.bbsrc.tgac.miso.core.data.workflow;
 
+import java.util.List;
+
 public interface Workflow {
+  Progress getProgress();
+
+  void setProgress(Progress progress);
+
+  WorkflowStepPrompt getNextStep();
+
+  WorkflowStepPrompt getStep(int stepNumber);
+
+  boolean isComplete();
+
+  List<String> getLog();
+
+  void processInput(ProgressStep step);
+
+  void processInput(int stepNumber, ProgressStep step);
```

Comment:
Javadoc to clarify that this can be a previously-completed step, and it is up to the implementation to invalidate later steps if necessary